src/components/codeContext.tsx

```diff
@@ -191,23 +190,71 @@ export async function fetchCodeKeywords(): Promise<CodeKeywords> {
   };
 }
 
+function getCsrfToken(): string | null {
+  // is sentry-sc in production, but may also be sc in other envs
+  // So we just try both variants
+  const cookieNames = ['sentry-sc', 'sc'];
+
+  return cookieNames
+    .map(cookieName => Cookies.get(cookieName))
+    .find(token => token !== null);
+}
+
+export async function createOrgAuthToken({
+  orgSlug,
+  name,
+}: {
+  name: string;
+  orgSlug: string;
+}) {
+  const baseUrl =
+    process.env.NODE_ENV === 'development'
+      ? 'http://dev.getsentry.net:8000/'
+      : 'https://sentry.io';
+
+  const url = `${baseUrl}/api/0/organizations/${orgSlug}/org-auth-tokens/`;
+
+  const body = {name};
+
+  try {
+    const resp = await fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(body),
+      credentials: 'include',
+      headers: {
+        Accept: 'application/json; charset=utf-8',
+        'Content-Type': 'application/json',
+        'X-CSRFToken': getCsrfToken(),
+      },
+    });
+
+    if (!resp.ok) {
+      return null;
+    }
+
+    const json = await resp.json();
+
+    return json.token;
+  } catch {
+    return null;
+  }
```

Comment:
will Sentry catch a bug here, in case it happens?

Reply:
No, we just ignore it in this case - not sure if it is worth it to capture this? May lead to some noise, not sure.